src/test/run-make/crate-attributes-with-cfg_attr/lib.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![cfg_attr(foo, crate_type="lib")]
```

Comment:
Ah I think an auxiliary file won't actually work due to the compiler passing `--crate-type`, but this should work as a run-pass test because the compiler only assumes a binary output if no other is specified:

```rust
// src/test/run-pass/foo.rs

// compile-flags: --cfg foo
#![crate_type = "lib"]
#![cfg_attr(foo, crate_type = "bin")]
extern crate bar;
fn main() {}
```